docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,98 @@
+---
+title: add-discovery-kernel-arguments
+authors:
+- "@oamizur"
+  creation-date: 2022-09-20
+  last-updated: 2022-10-23
+---
+
+# Add discovery kernel arguments
+
+## Summary
+
+Some environments require additional kernel arguments to boot host machines.  
+We would like to support additional kernel arguments while booting the discovery image.
+
+## Motivation
+
+Previously, adding kernel arguments was required by several customers in order to adapt to various environments.
+For example **rd.net.timeout.carrier** may be required in order for either static networking to be applied or 
+to successfully get a dhcp address before pulling the rootfs.
+### Goals
+Add capability to set additional kernel arguments to discovery image.
+
+- Image types: Full ISO, minimal ISO, iPXE
+- Deployment types: Cloud, operator
+
+### Non-Goals
+Manipulation of existing kernel parameters (update/delete).  Update and delete of exiting
+kernel arguments in ISO file is supported by Coreos Installer, but this implementation will not support it. 
```

Comment:
Why not?

Reply:
We decided this during the design meeting.

Reply:
IIRC we said that we *could* defer it as long as adding it later won't break the current API.  Will it?

Reply:
I don't recall that we said it will not break the current API. As I see it now, the API will have to change in order to support it.

Reply:
We can't plan to have API breakages.  It's fine if we don't implement update/delete right now, but we must be able to add them without breaking users.  This is a blocker as far as I'm concerned.

Reply:
FWIW this API was already merged in https://github.com/openshift/assisted-service/pull/4402 and implemented in https://github.com/openshift/assisted-service/pull/4425

Is there something we want to do now to prevent this from getting to stage/prod or are we okay just as long as it doesn't get in the UI?

Reply:
Changed